### PR TITLE
Updated the cypher query functions predicates page

### PR DESCRIPTION
**Pull request by martin-neotech**
Improved the text and examples.

### CODE REVIEW

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 1]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-          |AND all(x IN nodes(p) WHERE x.age > 30)
-          |RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `all()` returns `true` if the predicate holds for all elements in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("all(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
          #However, an implicit conversion will happen for single elements when passing node properties during cypher execution.""".stripMargin('#')),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 2]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-          |AND any(x IN a.array WHERE x = 'one')
-          |RETURN a.name, a.array""".stripMargin, ResultAssertions(r => {
+      p("""The function `any()` returns `true` if the predicate holds for at least one element in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("any(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
          #However, an implicit conversion will happen for single elements when passing node properties during cypher execution.""".stripMargin('#')),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 3]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
+        ("predicate", "A predicate that is tested against all items in the list."))
+      query("""MATCH (a)
+              #WHERE any(color IN a.liked_colors WHERE color = 'yellow')
+              #RETURN a.name, a.liked_colors""".stripMargin('#'),
+      ResultAssertions(r => {
           r.toList.length should equal(1)
           r.columnAs[String]("a.name").toList.head should equal("Eskil")
         })) {
-        p("All nodes in the returned paths have at least one *'one'* value set in the array property named `array`.")
+        p("The nodes returned have a property `liked_colors` that is a list containing at least one value with the value `'yellow'`.")
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        p("The query returns nodes with a list of their properties, where at least one property has the value `'yellow'`.")
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 4]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-        " `null` is returned if the list is `null` or all of its elements are `null`.")
-      function("none(variable IN list WHERE predicate)", "A Boolean.", ("list", "An expression that returns a list. "
-        +
-         "A single element cannot be explicitly passed as a literal in the cypher statement. " +
-         "However, an implicit conversion will happen for a single elements when passing node properties during cypher execution."), ("variable", "This is the variable that can be used from within the predicate."), ("predicate", "A predicate that is tested against all items in the list."))
-      query(
-        """MATCH p = (n)-[*1..3]->(b)
-          |WHERE n.name = 'Alice'
-          |AND none(x IN nodes(p) WHERE x.age = 25) RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `none()` returns `true` if the predicate holds for no element in the given list.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
      p("""The function `none()` returns `true` if the predicate does _not_ hold for any element in the given list.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 5]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("none(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
          #However, an implicit conversion will happen for single elements when passing node properties during cypher execution.""".stripMargin('#')),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 6]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
     section("single()", "functions-single") {
-      p("`single()` returns true if the predicate holds for exactly one of the elements in the given list." +
-        " `null` is returned if the list is `null` or all of its elements are `null`.")
-      function("single(variable IN list WHERE predicate)", "A Boolean.", ("list", "An expression that returns a list."), ("variable", "This is the variable that can be used from within the predicate."), ("predicate", "A predicate that is tested against all items in the list."))
-      query(
-        """MATCH p = (n)-->(b)
-          |WHERE n.name = 'Alice'
-          |AND single(var IN nodes(p) WHERE var.eyes = 'blue')
-          |RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `single()` returns `true` if the predicate holds for exactly one of the elements in the given list.
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
      p("""The function `single()` returns `true` if the predicate holds for exactly _one_ of the elements in the given list.
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 7]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-          |RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `all()` returns `true` if the predicate holds for all elements in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("all(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
+        ("variable", "This is the variable that can be used from within the predicate."),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        ("variable", "A variable that can be used from within the predicate."),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 8]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-          |RETURN a.name, a.array""".stripMargin, ResultAssertions(r => {
+      p("""The function `any()` returns `true` if the predicate holds for at least one element in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("any(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
+        ("variable", "This is the variable that can be used from within the predicate."),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        ("variable", "A variable that can be used from within the predicate."),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 9]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-          |AND none(x IN nodes(p) WHERE x.age = 25) RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `none()` returns `true` if the predicate holds for no element in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("none(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list",
+        """An expression that returns a list.
+          #A single element cannot be explicitly passed as a literal in the cypher statement.
+          #However, an implicit conversion will happen for a single elements when passing node properties during cypher execution.""".stripMargin('#')),
+        ("variable", "This is the variable that can be used from within the predicate."),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        ("variable", "A variable that can be used from within the predicate."),
```

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 10]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
-        """MATCH p = (n)-->(b)
-          |WHERE n.name = 'Alice'
-          |AND single(var IN nodes(p) WHERE var.eyes = 'blue')
-          |RETURN p""".stripMargin, ResultAssertions(r => {
+      p("""The function `single()` returns `true` if the predicate holds for exactly one of the elements in the given list.
+          #`null` is returned if the list is `null` or all of its elements are `null`.""".stripMargin('#'))
+      function("single(variable IN list WHERE predicate)",
+        "A Boolean.",
+        ("list", "An expression that returns a list."),
+        ("variable", "This is the variable that can be used from within the predicate."),
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        ("variable", "A variable that can be used from within the predicate."),
```

**Review by AlexicaWright — COMMENTED** [review]
Some comments and suggestions, but overall, it looks really good! Thank you @martin-neotech !

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala (hunk 3 above)
Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: I changed the query to return the node instead. And changed the text a little also.

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala (hunk 4 above)
Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: Nice spotted!

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 11]
```diff
@@ -81,17 +81,17 @@ class PredicateFunctionsTest extends DocumentingTest {
...
+      query("""MATCH (n)
+              #WHERE any(color IN n.liked_colors WHERE color = 'yellow')
+              #RETURN n""".stripMargin('#'),
       ResultAssertions(r => {
           r.toList.length should equal(1)
-          r.columnAs[String]("a.name").toList.head should equal("Eskil")
+          //r.columnAs[String]("n.name").toList.head should equal("Eskil")
         })) {
-        p("The nodes returned have a property `liked_colors` that is a list containing at least one value with the value `'yellow'`.")
+        p("The query return nodes with the property `liked_colors` with the value of a list containing at least one element with the value `'yellow'`.")
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: May I suggest: "The query returns nodes with the property `liked_colors`, as a list, where at least one element has the value `'yellow'`.

**Review by AlexicaWright — COMMENTED** [review]
Looks good, just one comment. Thank you @martin-neotech !

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 12]
```diff
@@ -91,7 +91,7 @@ class PredicateFunctionsTest extends DocumentingTest {
           r.toList.length should equal(1)
           //r.columnAs[String]("n.name").toList.head should equal("Eskil")
         })) {
-        p("The query return nodes with the property `liked_colors` with the value of a list containing at least one element with the value `'yellow'`.")
+        p("The query return nodes with the property `liked_colors` (a list), where at least one element has the value `'yellow'`.")
```
Comment (AlexicaWright) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: The query returnS nodes with the property `liked_colors`(AS a list), ....
(capitals just for emphasis :) )

**Review by AlexicaWright — COMMENTED** [review]
Just a small comment. Sorry I'm so picky :)

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala (hunk 12 above)
Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: updated

**Review by AlexicaWright — APPROVED** [review]
Looks fantastic! Thank you @martin-neotech

File: cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala [hunk 13]
```diff
@@ -27,124 +27,174 @@ class PredicateFunctionsTest extends DocumentingTest {
...
+              #  AND single(var IN nodes(p) WHERE var.eyes = 'blue')
+              #RETURN p""".stripMargin('#'),
+      /*Result:
+      ({name: "Alice", eyes: "brown", age: 38})-[:KNOWS]->({name: "Bob", eyes: "blue", age: 25})
+      */
+      ResultAssertions(r => {
           r.toList.length should equal(1)
         })) {
-        p("Exactly one node in every returned path has the `eyes` property set to *'blue'*.")
+        p("In every returned path there is exacly one node that has a property `eyes` with the value `'blue'`.")
```
Comment (recrwplay) [inline cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]:
```suggestion
        p("In every returned path there is exactly one node that has a property `eyes` with the value `'blue'`.")
```

Comment (martin-neotech) [reply cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/PredicateFunctionsTest.scala]: I need to run a spell checker next time :)

**Pull request merged**